src/components/types.ts

```diff
@@ -32,7 +32,9 @@ export interface InputProps extends InputStatus, InputHTMLAttributes<HTMLInputEl
    */
   endIcon?: ReactNode;
 
-  weight?: number;
+  fontWeight?: keyof AppTheme['typography']['weights'];
```

Comment:
The type of `keyof AppTheme['typography']['weights'];` this is an object not a string or a number?

https://github.com/zopaUK/react-components/blob/master/src/components/styles/Theme/ThemeProvider.tsx#L240-L245